src/elements/Button.tsx

```diff
@@ -0,0 +1,144 @@
+import { FunctionComponent } from "react";
+import {
+  CircularProgress,
+  Button as MUIButton,
+  ButtonProps as MUIButtonProps,
+} from "@mui/material";
+import theme from "theme";
+
+interface ButtonProps extends Omit<MUIButtonProps, "color" | "variant"> {
+  readonly color?:
+    | "company"
+    | "crypto"
+    | "magazine"
+    | "music"
+    | "partners"
+    | "white";
+  readonly isLoading?: boolean;
+  readonly variant?: "primary" | "secondary" | "outlined";
+  readonly width?: "compact" | "default" | "full" | "icon";
+}
+
+/**
+ * Implements Material UI Button.
+ * See https://mui.com/material-ui/api/button/ for extra props.
+ * Native `button` and `a` element attributes are also available to use.
+ */
+const Button: FunctionComponent<ButtonProps> = ({
+  children,
+  color = "music",
+  disabled = false,
+  isLoading = false,
+  variant = "primary",
+  width = "default",
+  ...rest
+}) => {
+  const widthStyles = {
+    compact: {
+      maxWidth: "max-content",
+    },
+    default: {
+      width: "100%",
+      maxWidth: "340px",
+    },
+    full: {
+      width: "100%",
+    },
+    icon: {
+      height: "40px",
+      minWidth: "unset",
+      p: 0,
+      width: "40px",
+    },
+  };
+
+  const variantStyles = {
+    primary: {
+      background:
+        color === "white" ? theme.gradients.music : theme.gradients[color],
```

Comment:
Since we dont have a "white" linear gradient for our primary buttons, i set it to default to our music gradient